src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -205,7 +265,10 @@ public void StopProcessingNextRequest()
                         stream.Abort(connectionError);
                     }
 
-                    await _frameWriter.WriteGoAwayAsync(_highestOpenedStreamId, errorCode);
+                    if (previousConnectionState != ConnectionState.Closed)
+                    {
+                        await _frameWriter.WriteGoAwayAsync(_highestOpenedStreamId, errorCode);
+                    }
```

Comment:
Super nit: add extra newline after `}` unless it's followed by an else/catch/finally or another `}`. This applies to a lot of other lines in this PR too.